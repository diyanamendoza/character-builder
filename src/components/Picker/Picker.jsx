import React from 'react';
import './Picker.css';

export default function Picker({
  head,
  setHead,
  torso,
  setTorso,
  feet,
  setFeet,
}) {
  const heroes = ['hulk', 'deadpool', 'spiderman', 'captainmarvel'];

  return (
    <div
      className="picker"
      style={{
        display: 'flex',
        flexDirection: 'column',
        gap: 10,
        marginBottom: 10,
      }}
    >
      <label>
        Head
        <select value={head} onChange={(e) => setHead(e.target.value)}>
          {heroes.map((pic) => {
            return <option key={pic}>{pic}</option>;
          })}
        </select>
      </label>
      <label>
        Torso
        <select value={torso} onChange={(e) => setTorso(e.target.value)}>
          {heroes.map((pic) => {
            return <option key={pic}>{pic}</option>;
          })}
        </select>
      </label>
      <label>
        Feet
        <select value={feet} onChange={(e) => setFeet(e.target.value)}>
          {heroes.map((pic) => {
            return <option key={pic}>{pic}</option>;
          })}
        </select>
      </label>
    </div>
  );
}
