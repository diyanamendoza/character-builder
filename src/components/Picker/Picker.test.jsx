import { screen, render } from '@testing-library/react';
import Picker from './Picker.jsx';

it('should render picker component', () => {
  const { container } = render(<Picker />);
  const selectHead = screen.getByLabelText(/Head/);
  screen.debug();
  expect(selectHead).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
