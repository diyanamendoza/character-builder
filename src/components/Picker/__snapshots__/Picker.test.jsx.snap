// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render picker component 1`] = `
<div>
  <div
    class="picker"
    style="display: flex; flex-direction: column; gap: 10px; margin-bottom: 10px;"
  >
    <label>
      Head
      <select>
        <option>
          hulk
        </option>
        <option>
          deadpool
        </option>
        <option>
          spiderman
        </option>
        <option>
          captainmarvel
        </option>
      </select>
    </label>
    <label>
      Torso
      <select>
        <option>
          hulk
        </option>
        <option>
          deadpool
        </option>
        <option>
          spiderman
        </option>
        <option>
          captainmarvel
        </option>
      </select>
    </label>
    <label>
      Feet
      <select>
        <option>
          hulk
        </option>
        <option>
          deadpool
        </option>
        <option>
          spiderman
        </option>
        <option>
          captainmarvel
        </option>
      </select>
    </label>
  </div>
</div>
`;
